feat: Adds Stream Instance L1 CDK construct (#233)
mongodb/awscdk-resources-mongodbatlas: src/l1-resources/stream-instance/README.md

# stream-instance

The official [MongoDB Atlas](https://www.mongodb.com/) AWS CDK resource for Node.js.

> AWS CDK [L1 construct] and data structures for the [AWS CloudFormation Registry] type `MongoDB::Atlas::StreamInstance`.
[L1 construct]: https://docs.aws.amazon.com/cdk/latest/guide/constructs.html
[AWS CloudFormation Registry]: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/registry.html

## Description

> Returns, adds, edits, and removes stream instances.

Reviewer: We should call out in all CDK documentation as well that this is in public preview like we did in CFN. 

Also, we should at least include reference to this README https://github.com/mongodb/mongodbatlas-cloudformation-resources/blob/master/examples/atlas-streams/README.md 
cc @Zuhairahmed in case he wants to add anything.

Author: I added the same as we did in CFN to make explicit that it's in preview.

Reviewer: I was also referring to the https://github.com/mongodb/mongodbatlas-cloudformation-resources/blob/master/examples/atlas-streams/README.md because it also mentions about Stream processors.

We need to make sure all relevant documentations points from the Scope doc are addressed.

## MongoDB Atlas API Docs

For more information about the API refer to: [API Endpoints](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Streams)

## Usage

In order to use this library, you will need to activate this AWS CloudFormation Registry type in your account. You can do this via the AWS Management Console or using the [AWS CLI](https://aws.amazon.com/cli/) using the following command:

```sh
aws cloudformation activate-type \

>   --type-name MongoDB::Atlas::SearchDeployment \

Reviewer 2: ```suggestion
  --type-name MongoDB::Atlas::StreamInstance \
```

  --publisher-id bb989456c78c398a858fef18f2ca1bfc1fbba082 \
  --type RESOURCE \
  --execution-role-arn ROLE-ARN
```

Alternatively:

```sh
aws cloudformation activate-type \

>   --public-type-arn arn:aws:cloudformation:us-east-1::type/resource/bb989456c78c398a858fef18f2ca1bfc1fbba082/MongoDB-Atlas-SearchDeployment \

Reviewer 2: ```suggestion
  --public-type-arn arn:aws:cloudformation:us-east-1::type/resource/bb989456c78c398a858fef18f2ca1bfc1fbba082/MongoDB-Atlas-StreamInstance \
```

  --execution-role-arn ROLE-ARN
```

You can find more information about activating this type in the [AWS CloudFormation documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/registry-public.html).

## Example: [stream-instance.ts](../../../examples/l1-resources/stream-instance.ts)
```ts

> import { CfnSearchDeployment } from 'awscdk-resources-mongodbatlas';

Reviewer 2: ```suggestion
import { CfnStreamInstance } from 'awscdk-resources-mongodbatlas';
```

    const streamInstance = new CfnStreamInstance(this, "stream-instance-testing-stack", {
      profile: atlasProps.profile,
      instanceName: atlasProps.projectId,
      projectId: atlasProps.projectId,
      dataProcessRegion: {
        cloudProvider: StreamsDataProcessRegionCloudProvider.AWS,
        region: "VIRGINIA_USA",
      },
      streamConfig: {
        tier: "SP30",
      },
    });
```

## Feedback

This library is auto-generated and published to all supported programming languages by the [cdklabs/cdk-cloudformation] project based on the API schema published for `MongoDB::Atlas::StreamInstance`.

> * Issues related to this generated library should be [reported here](https://github.com/cdklabs/cdk-cloudformation/issues/new?title=Issue+with+%40cdk-cloudformation%2Fmongodb-atlas-project+v1.0.0).

Reviewer: ```suggestion
* Issues related to this generated library should be [reported here](https://github.com/cdklabs/cdk-cloudformation/issues/new?title=Issue+with+%40cdk-cloudformation%2Fmongodb-atlas-stream-instance+v1.0.0).
```
